controllers/validations.go

```diff
@@ -66,6 +68,29 @@ func isManagedCluster(ctx context.Context, client client.Client, clusterName str
 	return nil
 }
 
+func isVersionCompatible(peerRef multiclusterv1alpha1.PeerRef, clientInfoMap map[string]string) error {
+	clientInfo, err := getClientInfoFromConfigMap(clientInfoMap, utils.GetKey(peerRef.ClusterName, peerRef.StorageClusterRef.Name))
+	if err != nil {
+		return fmt.Errorf("validation: unable to get client info: error: %v", err)
+	}
+	spokeVersion, err := semver.Parse(clientInfo.ProviderInfo.Version)
+	if err != nil {
+		return fmt.Errorf("validation: unable to parse spoke version: error: %v", err)
+	}
+	spokeVersion.Pre = nil
+	spokeVersion.Build = nil
+	hubVersion, err := semver.Parse(version.Version)
+	if err != nil {
+		return fmt.Errorf("validation: unable to parse hub version: error: %v", err)
+	}
+	hubVersion.Pre = nil
+	hubVersion.Build = nil
+	if !spokeVersion.Equals(hubVersion) {
+		return fmt.Errorf("storage cluster version %q on ManagedCluster %q is incompatible with Multicluster Orchestrator version %q", spokeVersion, peerRef.ClusterName, hubVersion)
```

Comment:
```suggestion
		return fmt.Errorf("StorageCluster version %q on ManagedCluster %q is incompatible with Multicluster Orchestrator version %q", spokeVersion, peerRef.ClusterName, hubVersion)
```

nit, we generally use full resource names

Reply:
I started with that, but when we return errors, it can not start with a capital letter.
Also, when I say "storage cluster" it means exactly that, but when I say "StorageCluster" it refers to the CR. In this case, we are not talking about the CR version.

Reply:
Ok,  I would prefer using CR as that may help redirect QE and devs to check those particular resources.  Possibly could help with metrics for log exploring tools. 

If you can, please reframe it